src/System.Net.Http/src/netcore50/System/Net/HttpClientHandler.cs

```diff
@@ -321,7 +321,10 @@ public long MaxRequestContentBufferSize
 
         public int MaxResponseHeadersLength
         {
-            get { return 0; } // TODO: Issue #8541 - return default value used in Wininet.
+            // Windows.Web.Http is built on WinINet. There is no maximum limit (except for out of memory)
+            // for received response headers. So, returning 0 (indicating no limit) is appropriate.
+            get { return 0; }
```

Comment:
I lean towards liking -1 as the value for unlimited, as it allows all non-negative values to have their numerical meaning, and it maps to how -1 is used elsewhere in the framework to represent unlimited, even if that's largely inspired by Win32 (though -1 is also frequently used as a sentinel on Unix as well, e.g. to indicate that a function call failed).